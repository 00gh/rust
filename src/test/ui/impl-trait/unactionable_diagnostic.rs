trait Trait {}

struct Foo;

impl Trait for Foo {}

fn foo<'t, P>(
    post: P,
    x: &'t Foo,
) -> &'t impl Trait {
    x
}

fn bar<'t, T>(
    post: T,
    x: &'t Foo,
) -> &'t impl Trait {
    foo(post, x)
    //~^ ERROR: the opaque type `foo<T>::{opaque#0}` may not live long enough
    //~| HELP: consider adding an explicit lifetime bound `foo<T>::{opaque#0}: 't`
}

fn main() {}
