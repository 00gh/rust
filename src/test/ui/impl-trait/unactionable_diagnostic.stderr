error[E0309]: the opaque type `foo<T>::{opaque#0}` may not live long enough
  --> $DIR/unactionable_diagnostic.rs:18:5
   |
LL |     foo(post, x)
   |     ^^^^^^^^^^^^
   |
   = help: consider adding an explicit lifetime bound `foo<T>::{opaque#0}: 't`...
   = note: ...so that the type `impl Trait` will meet its required lifetime bounds

error: aborting due to previous error

For more information about this error, try `rustc --explain E0309`.
